### ISSUE TITLE
[Help]: Sentinel XML EOF not well formatted ?? S1_TOPS_Frame.csh and  ext_orb_s1a

### ISSUE THREAD

**Issue by gg-bb** [gmtsar/gmtsar#986]
Good Morning everyone!

I want to process two sentinel 1A TOPS image to produce an interferogram. I use the "p2p_S1_TOPS_Frame.csh" to process the images following the example provided in the LosAngeles TOPS  test files (which works fine).

I have the 2 following SAFE data downloaded from Peps website (Marocco 2023 M7 earthquake, CNES data server):
master: S1A_IW_SLC__1SDV_20230810T183342_20230810T183410_049817_05FDB6_EA37.SAFE
slave:   S1A_IW_SLC__1SDV_20230903T183344_20230903T183412_050167_0609B4_100E.SAFE

with the two associated EOF files using the eof command line (see eof.download python tool):
master:  S1A_OPER_AUX_POEORB_OPOD_20230830T080726_V20230809T225942_20230811T005942.EOF
slave:     S1A_OPER_AUX_POEORB_OPOD_20230923T080643_V20230902T225942_20230904T005942.EOF

SAFE and EOF files are stored into the wdir/raw directory

the dem is generated from SRTM30  hgt files that are blended and cropped according to the image footprint and stored in wdir/topo. I tried my dem.grd generator with the LosAngeles TOPS data and it worked.

However, when i run the  command in my wdir (with complete file names ):

>p2p_S1_TOPS_Frame.csh master.SAFE master.EOF slave.SAFE slave.EOF config.txt vv 0

the following error message appears at the beginning of the process  (while processing subswath 1):

"...
S1_20230810_183342_F1
S1_20230903_183344_F1
Reading in 6133 lines of info from XML...
Writing 17 lines for orbit...
Reading in 6119 lines of info from XML...
Writing 17 lines for orbit...
+++++++++++++++++++++++++++++++++++++++
Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file...
Not an well formatted XML file...-1, -1
Segmentation fault (core dumped)
+++++++++++++++++++++++++++++++++++++++
..."

After a little digging I tried the ext_orb_s1a command with the first subswath  directly in the wdir/F1/raw/ directory:

>ext_orb_s1a  S1_master_F1.PRM master.EOF test

and i obtained the results:

"Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file..."

I have tried with an another area with recent data (acquired in 2024) and I obtained the same issue. It is like the EOF xml file are somehow corrupted which is surprising as i didn't edit them and I download them from the "source" (ie copernicus scihub). I compared the EOF files given in the LosAngeled TOPS example but i didn't find any significant differences.

How can i solve this issue? Any help would highly appreciated

Have a nice day,

**OS and software version**
OS: Ubuntu 22.04.4 LTS
GMTSAR version: >6

**Comment by gg-bb** [comment]
Any help?
Couldn't figure out the problem..
Is it a bug?
Best regards
GB

**Comment by AlexeyPechnikov** [comment]
First, retrieve the orbits from the known storage.​ The easiest method is to use this auto-updating GitHub repository, which efficiently manages Sentinel-1 orbits by date: https://github.com/AlexeyPechnikov/S1orbits.

**Comment by gg-bb** [comment]
Thank You Alexey for your help.
I reprocessed the data using the EOF file downloaded using the link you gave me. 
It didn't worked and that's really odd as the example of LosAngeles works just fine.
Is there any issue with the xml lib of GMTSAR?


I used the following command:

>p2p_S1_TOPS_FRAME.csh
S1A_IW_SLC__1SDV_20230810T183342_20230810T183410_049817_05FDB6_EA37.SAFE S1A_OPER_AUX_POEORB_OPOD_20230830T080726_V20230809T225942_20230811T005942.EOF S1A_IW_SLC__1SDV_20230903T183344_20230903T183412_050167_0609B4_100E.SAFE S1A_OPER_AUX_POEORB_OPOD_20230923T080643_V20230902T225942_20230904T005942.EOF 
config.txt 
vv 
1


However, i obtained the same results (log F1):

>PREPROCESS - START
Working on images s1a-iw1-slc-vv-20230810t183342-20230810t183410-049817-05fdb6-004 s1a-iw1-slc-vv-20230903t183344-20230903t183412-050167-0609b4-004 ...
rm: No match.
rm: No match.
rm: No match.
pre_proc.csh
 Pre-Process S1_TOPS data - START
ln: failed to create symbolic link './dem.grd': File exists
grep: No match.
grep: No match.
grep: No match.
S1_20230810_183342_F1
S1_20230903_183344_F1
Reading in 6133 lines of info from XML...
Writing 17 lines for orbit...
Reading in 6119 lines of info from XML...
Writing 17 lines for orbit...
++++++++++++++modification GB++++++++++++++++
s1a-iw1-slc-vv-20230810t183342-20230810t183410-049817-05fdb6-004.EOF
Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file...
Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file...
s1a-iw1-slc-vv-20230903t183344-20230903t183412-050167-0609b4-004.EOF
+++++++++++++++++++++++++++++++++++++++++++++
Successfully opened S1_20230810_183342_F1.PRM 
interpolation point outside of data constraints, 19161222.788730 0.000000 0.000000
cat: tmp: No such file or directory
grep: tmp: No such file or directory"_

**Comment by AlexeyPechnikov** [comment]
It works on my end with the specified scenes and orbit files you listed above:

![image](https://github.com/user-attachments/assets/d7661ffe-5e75-4d24-a0a9-cbf2154fe055)

![image](https://github.com/user-attachments/assets/34beac65-7ac2-403e-a569-5c8cdd4bc467)

Please wait for a response from @Xiaohua-Eric-Xu to help resolve the issue.​

**Comment by gg-bb** [comment]
Nice Job,
Ok, so it is very likely coming from my installation... Hopefully I won't have to reinstall everything.
Many Thanks for your support, 
I am looking forward any additional insights from @Xiaohua-Eric-Xu

**Comment by gg-bb** [comment]
Hi,
Ok, it seems to work for me now with the above dataset. I re-install GMT, and GMTSAR with the "-z muldefs" option added to the config.mk (CFLAGS and LDFLAGS - step 4).
I believe that after I upgraded my ubuntu 20 to 22 (with GMTSAR already installed)   dependencies were broken somehow.
The weird point is that I could run the LA example without issue with my corrupted installation.

For the record here is how I proceeded to fix my problem:
-remove my old GMTSAR
-sudo apt-get update; sudo apt-get upgrade (Ubuntu 22)
-Then follow the GMT/GMTSAR installation procedure with the "-z" option
 
Thank for your support,
Best Regards.
